mobile/src/components/popups/OneButtonPopup.tsx

```diff
@@ -0,0 +1,45 @@
+import { Alert, Modal, Text, Pressable, View } from "react-native";
+
+import { AppStylesheet } from "../../styles/AppStylesheet";
+
+interface OneButtonPopupProps {
+  labelText: string;
+  buttonText: string;
+  buttonFunc: () => void;
+}
+
+// Pass in as props the button's label text, button text, and what the button does when pressed
+const OneButtonPopup = ({ labelText, buttonText, buttonFunc }: OneButtonPopupProps) => (
+  <Modal
+    animationType="slide"
+    transparent
+    onRequestClose={() => {
+      Alert.alert("Modal has been closed.");
+    }}
+  >
+    <View style={AppStylesheet.centeredView}>
+      <View style={AppStylesheet.modalView}>
+        <Text style={AppStylesheet.modalText}>{labelText}</Text>
+        <View style={AppStylesheet.buttonContainer}>
+          {buttonText.length > 10 ? (
+            <Pressable
+              style={[AppStylesheet.modalButton, { width: 150, marginTop: 40 }]}
```

Comment:
Why are these extra width/marginTop styles not in the stylesheet?

Reply:
It's because given the size of the text, modalButton needs to change it's width to fit the text in the button. The format above overrides the width and margin top of the modalButton style to accomodate. I tried to make it a stylesheet prop but it meant that I had to pass it in every time I used any style from the sheet. I also tried to make unique props for only this one style but typescript was complaining. Is there another way to put this into the stylesheets?

Reply:
I think this is fine since this will have to be dynamic depending on how much text is on the button. I don't think we want to hardcode this for every popup.